scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -356,7 +356,7 @@ class Router(formatOps: FormatOps) {
 
       case FormatToken(T.RightArrow() | T.ContextArrow(), right, _)
           if leftOwner.is[Term.FunctionTerm] ||
-            (leftOwner.is[Template] &&
+            leftOwner.is[Term.PolyFunction] || (leftOwner.is[Template] &&
```

Comment:
completely optional: the new term looks to go better, logically, with the previous line:
```
if leftOwner.is[Term.FunctionTerm] || leftOwner.is[Term.PolyFunction] ||
  (leftOwner.is[Template] && // unchanged
```
if it fits, of course.

Reply:
The if line is still to long to changed it to:
```scala
 if leftOwner.is[Term.FunctionTerm] ||
   leftOwner.is[Term.PolyFunction] ||      
  (leftOwner.is[Template] &&
```